### PR TITLE
[WIP] add rh sub

### PR DESCRIPTION
**Pull request by twaldrop**

### CODE REVIEW

File: roles/common/tasks/main.yml [hunk 1]
```diff
@@ -20,6 +20,33 @@
...
+    password: "{{ redhat.pass }}"
+  when: ansible_distribution == "RedHat"
+
+- name: Add OSP pool
+  command: subscription-manager attach --pool={{ item }}
+  when: ansible_distribution == "RedHat"
+  with_items:
+  - 8a85f98158880f8d01588bde0f7c3c43
+
+- name: add OS respositories
```
Comment (zfeldstein) [inline roles/common/tasks/main.yml]: Can we add a sleep, or someway to validate if the host is registered? I think it takes around a minute after you register before you can add pools or repos which kills the ursula run.

Comment (twaldrop) [inline roles/common/tasks/main.yml]: updated

File: envs/example/allinone-centos/vagrant.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 default:
   memory: 512
   cpus: 1
-  box: centos/7
+  box: rhel7
```
Comment (nirajdp76) [inline envs/example/allinone-centos/vagrant.yml]: we should create new envs for allinone-rhel7 and leave allinone-centos operational for centos

File: roles/common/tasks/main.yml [hunk 3]
```diff
@@ -20,6 +20,37 @@
     retries: 5
   when: ansible_distribution == "Ubuntu"
 
+- name: Register with RHN
```
Comment (nirajdp76) [inline roles/common/tasks/main.yml]: Need to create a separate role for rhn-subscription and move logic of subscription, attaching to pool, disbale all repos )only flotsam) and enabled desire repos.

File: roles/common/tasks/main.yml [hunk 4]
```diff
@@ -20,6 +20,37 @@
...
+
+- name: Pause for registration
+  pause:
+    seconds: 90
+
+- name: Add OSP pool
+  command: subscription-manager attach --pool={{ item }}
+  when: ansible_distribution == "RedHat"
+  with_items:
+  - 8a85f98158880f8d01588bde0f7c3c43
```
Comment (nirajdp76) [inline roles/common/tasks/main.yml]: need to make this variable
redhat.pool_id.osp and redhat.pool_id.ceph

File: roles/common/tasks/main.yml [hunk 5]
```diff
@@ -20,6 +20,37 @@
...
+
+- name: add OS respositories
+  command: subscription-manager repos --enable={{ item }}
+  when: ansible_distribution == "RedHat"
+  with_items:
+  - rhel-7-server-extras-rpms
```
Comment (nirajdp76) [inline roles/common/tasks/main.yml]: need to make this variable
redhat.osp_repos and redhat.ceph_repos, again redhat.ceph_repos should be enabled only on osd node where we will also attach to ceph poolid.

Please discuss variable structure with @scoop206

**Comment by nirajdp76** [comment]
Just summarizing key points here:
1: we need a role that handles rhn-subscription logic
    1: using creds from variables subscribe, might need a sleep as subscription takes 60 secs or so
    2: At high level we deal with 2 set of artifacts. We need to make pool_id and repos variables. Pool_id and repos can change in future. When run in flotsam/bootstrap flow we need to also disable all repos before enabling desire repos.
        1: OSP entitlements which will be enabled on every node
              redhat.osp.pool_id: 8a85f98158880f8d01588bde0f7c3c43
              redhat.osp.repos: (list) rhel-7-server-rpms rhel-7-server-extras-rpms rhel-7-server-optional-rpms rhel-7-server-openstack-10-rpms  rhel-7-server-rh-common-rpms  rhel-7-server-openstack-10-tools-rpms rhel-7-server-openstack-10-optools-rpms rhel-7-server-openstack-10-tools-debug-rpms rhel-7-server-rhceph-2-mon-rpms rhel-7-server-rhceph-2-tools-rpms
        2: Ceph OSD  entitlements which needs to be enabled only on ceph osd nodes when ceph is enabled
             redhat.osd.pool_id:
             redhat.osd.repos: (list) rhel-7-server-rhceph-2-osd-rpms

File: envs/example/allinone-centos/vagrant.yml (hunk 2 above)
Comment (scoop206) [inline envs/example/allinone-centos/vagrant.yml]: ++

File: roles/common/tasks/main.yml (hunk 4 above)
Comment (paulczar) [inline roles/common/tasks/main.yml]: :+1:

File: roles/common/tasks/main.yml (hunk 1 above)
Comment (paulczar) [inline roles/common/tasks/main.yml]: we should probably try and validate after the sleep that it completed.

**Comment by scoop206** [comment]
+1 @nirajdp76 comments
Breaking out a new role makes sense to me given that we want a flexible var structure here

File: roles/common/tasks/main.yml (hunk 5 above)
Comment (paulczar) [inline roles/common/tasks/main.yml]: I think we need three lists "base" "osp" and "osd" that can be added as appropriate based on group membership ?

File: envs/example/allinone-rhel7/group_vars/all.yml [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+      - vim
+      - sysstat
+      - logrotate
+      - ntp
+      - lvm2
+      - cracklib
+      - mlocate
+  python:
+    system_packages:
+      - python-pip
```
Comment (chengtcli) [inline envs/example/allinone-rhel7/group_vars/all.yml]: Which yum repo is `python-pip` package in? I can find it even I enable all redhat repo mentioned in this PR.

**Comment by nirajdp76** [comment]
Closing this PR as is cover under new PR https://github.com/blueboxgroup/ursula/pull/2388

**Pull request closed without merging**